php-checks/src/main/java/org/sonar/php/checks/MultilineBlocksCurlyBracesCheck.java

```diff
@@ -0,0 +1,150 @@
+  name = "Multiline blocks should be enclosed in curly braces",
+  priority = Priority.CRITICAL,
+  tags = {Tags.BUG, Tags.CWE})
+@ActivatedByDefault
+@SqaleSubCharacteristic(SubCharacteristics.LOGIC_RELIABILITY)
+@SqaleConstantRemediation("5min")
+public class MultilineBlocksCurlyBracesCheck extends PHPSubscriptionCheck {
+
+  public static final String KEY = "S2681";
+  private static final String MESSAGE_LOOP = "Only the first line of this %s-line block will be executed in a loop. The rest will execute only once.";
+  private static final String MESSAGE_IF = "Only the first line of this %s-line block will be executed conditionally. The rest will execute unconditionally.";
+
+
+  private List<StatementTree> statements;
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    return CheckUtils.STATEMENT_CONTAINERS;
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    statements = CheckUtils.getStatements(tree);
+
+    for (int i = 0; i < statements.size() - 1; i++) {
+      StatementTree currentStatement = statements.get(i);
+
+      if (currentStatement.is(Kind.IF_STATEMENT)) {
+        checkStatement(getLastStatement((IfStatementTree)currentStatement), i, MESSAGE_IF);
+
+      } else if (currentStatement.is(Kind.FOR_STATEMENT)) {
+        checkStatement(((ForStatementTree) currentStatement).statements().get(0), i, MESSAGE_LOOP);
+
+      } else if (currentStatement.is(Kind.FOREACH_STATEMENT)) {
+        checkStatement(((ForEachStatementTree) currentStatement).statements().get(0), i, MESSAGE_LOOP);
+
+      } else if (currentStatement.is(Kind.WHILE_STATEMENT)) {
+        checkStatement(((WhileStatementTree) currentStatement).statements().get(0), i, MESSAGE_LOOP);
+
+      }
+
+    }
+  }
+
+  private static StatementTree getLastStatement(IfStatementTree ifStatement) {
+    if (ifStatement.elseClause() == null && ifStatement.elseifClauses().isEmpty()) {
+      return ifStatement.statements().get(0);
+    }
+
+    StatementTree statement = ifStatement;
+
+    do {
+      IfStatementTree nestedIfStatement = (IfStatementTree) statement;
+
+      if (nestedIfStatement.elseClause() != null) {
+        statement = nestedIfStatement.elseClause().statements().get(0);
+
+      } else {
+        List<ElseifClauseTree> elseifClauses = nestedIfStatement.elseifClauses();
+        if (elseifClauses.isEmpty()) {
+          statement = nestedIfStatement.statements().get(0);
+
+        } else {
+          statement = elseifClauses.get(elseifClauses.size() - 1).statements().get(0);
+        }
+      }
+    } while (statement.is(Kind.IF_STATEMENT));
+
+    return statement;
+  }
+
+  private void checkStatement(StatementTree firstInnerStatement, int nestingStatementNum, String message) {
+    if (firstInnerStatement.is(Kind.BLOCK)) {
+      return;
+    }
+
+    int firstIndent = column(firstInnerStatement);
+    List<StatementTree> shouldBeNested = new ArrayList<>();
+
+    for (int i = nestingStatementNum + 1; i < statements.size(); i++) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS135)